components/org.wso2.carbon.identity.application.authenticator.oidc/src/main/java/org/wso2/carbon/identity/application/authenticator/oidc/OpenIDConnectAuthenticator.java

```diff
@@ -526,10 +526,14 @@ protected void buildClaimMappings(Map<ClaimMapping, String> claims, Map.Entry<St
 
             }
         } catch (Exception e) {
-            claimValue = new StringBuilder(entry.getValue().toString());
+            if (log.isDebugEnabled()) {
+                log.debug("Error while mapping " + entry.getKey() + " with value " + entry.getValue(), e);
+            }
+            claimValue = entry.getValue() != null ? new StringBuilder(entry.getValue().toString()) : new StringBuilder();
         }
 
-        claims.put(ClaimMapping.build(entry.getKey(), entry.getKey(), null, false), claimValue.toString());
+        claims.put(ClaimMapping.build(entry.getKey(), entry.getKey(), null, false),
+                claimValue != null ? claimValue.toString() : null);
```

Comment:
Rather than having a 'null' for the default value, it would be better to use an empty string (StringUtils.EMPTY). Anyhow, when going over the code flow being improved, cannot observe a case where the claim value could be null. Yet, defining the default value for an empty string would be ideal as the test case covers for null values as well.